### PR TITLE
Replace RemoveAll with Remove

### PR DESCRIPTION
**Pull request by cdmihai**
Solves #2069 

Works around https://github.com/dotnet/corefx/issues/20609

### CODE REVIEW

**Comment by rainersigwald** [comment]
@dotnet-bot test Windows_NT Build for Full please

**Comment by rainersigwald** [comment]
@dotnet-bot test Windows_NT Build for Full please

File: src/Build.UnitTests/Evaluation/ItemEvaluation_Tests.cs [hunk 1]
```diff
@@ -507,5 +507,17 @@ public void LongIncludeChain()
...
+        // see https://github.com/Microsoft/msbuild/issues/2069
+        [Fact]
+        public void ImmutableListBuilderBug()
+        {
+            var content = @"<i Include=""0;x1;x2;x3;x4;x5;6;7;8;9""/>
+                            <i Remove=""x*""/>";
+
+            IList<ProjectItem> items = ObjectModelHelpers.GetItemsFromFragment(content);
+
+            Assert.Equal("0;6;7;8;9", String.Join(";", items.Select(i => i.EvaluatedInclude)));
```
Comment (rainersigwald) [inline src/Build.UnitTests/Evaluation/ItemEvaluation_Tests.cs]: There's an `Assert.Equal<T>(IEnumerable<T> expected, IEnumerable<T> actual, IEqualityComparer<T> comparer)` that gives nice error messages for this situation.

**Comment by cdmihai** [comment]
@dotnet-bot test Windows_NT Build for Full please

**Pull request merged**